app/controllers/manage_proposal_controller.rb

```diff
@@ -166,4 +166,26 @@ def submitter_from_proposal(proposal)
     proposal_urs_user = retrieve_urs_users([proposal['submitter_id']])[0]
     proposal_urs_user.blank? ? nil : { name: "#{proposal_urs_user['first_name']} #{proposal_urs_user['last_name']}", email: proposal_urs_user['email_address'] }
   end
+
+  # This function is mirrored in the proposal draft controller to sort an
+  # array of records instead of an array of hashes. They should be kept similar
+  # to maintain consistency in behavior for the proposal workflow.
+  def sort_hashes_by_submitter(proposals,user_hash = {})
+    @query = {}
+    @query['sort_key'] = params['sort_key'] unless params['sort_key'].blank?
+
+    sorted_proposals = proposals.sort do |a, b|
+                         # Making these arrays allows empty submitter_ids to sort last/first for ASC/DESC
+                         # Using 0 for 'has submitter id' and 1 for 'does not have submitted id'
+                         # allows these to naturally sort last in ASC order, and first in DESC order
+                         a_name = user_hash[a['submitter_id']] ? [0, user_hash[a['submitter_id']]] : [1, user_hash[a['submitter_id']]]
```

Comment:
I wonder if you could reuse the same function, and just have an "if" statement here (pseudocode)
a_submitted_id = if a instance_of hash ? a['submitted_id'] : a.submitted_id
b_submitted_id = if b instance_of hash ? b['submitted_id'] : b.submitted_id

Reply:
I implemented this.  I'm not fond of instance_of, but I prefer maintaining the logic once.